main.go

```diff
@@ -714,13 +709,9 @@ func (tf *transformer) handleDirectives(comments []*ast.CommentGroup) {
 				continue
 			}
 			// This directive has two arguments: "go:linkname localName newName"
-			localName := fields[1]
 
-			// The local name must not be obfuscated.
-			obj := tf.pkg.Scope().Lookup(localName)
-			if obj != nil {
-				tf.ignoreObjects[obj] = true
-			}
+			// obfuscate the local name.
+			fields[1] = hashWith(curPkg.GarbleActionID, fields[1])
```

Comment:
This passes tests, but doesn't quite sit right with me... my gut tells me I should add a check to see if the localName is ignored due to reflection. Fuctions can be reflected, so it's an edge case but possible I think.

Reply:
@mvdan what do you think?

Reply:
We only care about names. The only names that can show up in reflection types are:

* The name of a named type
* The name of fields in a struct type

A function name is never in there, I'm pretty sure. Even if you did `reflect.TypeOf(someFuncDecl)`, you'd get the function signature - it's not a named type in any way.

You could have `type someNamedFunc func(...)`, but then that's just a regular old named type, and already covered by our existing code.

Reply:
Ok, so this code should be fine then?

Reply:
Yep, I think it's fine. We always obfuscate the names of standalone functions.